src/main/java/be/syntra/devshop/DevshopBack/services/SearchServiceImpl.java

```diff
@@ -22,61 +30,116 @@ public SearchServiceImpl(ProductService productService) {
-
-        if (StringUtils.hasText((searchModel.getSearchRequest()))) {
-            productList = productList.parallelStream()
-                    .filter(product -> product.getName()
-                            .toLowerCase()
-                            .contains(searchModel.getSearchRequest().toLowerCase()))
-                    .collect(Collectors.toUnmodifiableList());
+    public ProductPageAndMinMaxPrice applySearchModel(SearchModel searchModel) {
+        log.info("searchModel -> {}", searchModel);
+        Pageable pageable = getSortingPageable(setDefaultPaginationValues(searchModel));
+        if (searchModel.isSearchResultView()) {
+            if (StringUtils.hasText(searchModel.getSearchRequest())) {
+                if (isPriceFilterActiveAndValid(searchModel)) {
+                    return getSearchResultsAndMinMaxPrice(
+                            productService.findAllNonArchivedBySearchTermAndPriceBetween(
+                                    searchModel.getSearchRequest(), searchModel.getPriceLow(), searchModel.getPriceHigh(), pageable),
+                            productService.findMinPriceProductNonArchivedBySearchTermAndPriceBetween(
+                                    searchModel.getSearchRequest(), searchModel.getPriceLow(), searchModel.getPriceHigh()),
+                            productService.findMaxPriceProductNonArchivedBySearchTermAndPriceBetween(
+                                    searchModel.getSearchRequest(), searchModel.getPriceLow(), searchModel.getPriceHigh())
+                    );
+                } else {
+                    return getSearchResultsAndMinMaxPrice(
+                            productService.findAllByNameContainingIgnoreCaseAndArchivedFalse(
+                                    searchModel.getSearchRequest(), pageable),
+                            productService.findMinPriceProductByNameContainingIgnoreCaseAndArchivedFalse(
+                                    searchModel.getSearchRequest()),
+                            productService.findMaxPriceProductByNameContainingIgnoreCaseAndArchivedFalse(
+                                    searchModel.getSearchRequest())
+                    );
+                }
+            }
+            if (StringUtils.hasText(searchModel.getDescription())) {
+                if (isPriceFilterActiveAndValid(searchModel)) {
+                    return getSearchResultsAndMinMaxPrice(
+                            productService.findAllNonArchivedByDescriptionAndPriceBetween(searchModel.getDescription(), searchModel.getPriceLow(), searchModel.getPriceHigh(), pageable),
+                            productService.findMinPriceProductNonArchivedByDescriptionAndPriceBetween(searchModel.getDescription(), searchModel.getPriceLow(), searchModel.getPriceHigh()),
+                            productService.findMaxPriceProductNonArchivedByDescriptionAndPriceBetween(searchModel.getDescription(), searchModel.getPriceLow(), searchModel.getPriceHigh())
```

Comment:
First you place the parameters on the next line. And from here you one-line them?
Also (more a complaint about github ;) ) I had to shift+scroll since the page is so long.

Reply:
forgot to clean it up, it's better now